model/src/main/scala/fr/iscpif/schelling/quantity/initial/PrefAttDiffusionContainer.scala

```diff
@@ -0,0 +1,114 @@
+package fr.iscpif.schelling.quantity.initial
+
+import fr.iscpif.schelling.quantity.Cell
+
+import scala.util.Random
+
+trait PrefAttDiffusionContainer <: Container {
+
+    /** sum of all capacities */
+    def totalCapacity : Double
+
+    /** diffusion parameters */
+    def diffusion : Double
+    def diffusionSteps : Int
+
+    /** Growth rate */
+    def growthRate : Int
+
+    /** Preferential attachment parameter */
+    def alphaAtt : Double
+
+
+
+
+    var values : Seq[Seq[Cell]] = null
```

Comment:
This should be the return of the function container, not a shared mutable value.